src/event/event.ts

```diff
@@ -19,13 +19,16 @@ export interface NodeEventContext {
 export class H3Event<
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   _RequestT extends EventHandlerRequest = EventHandlerRequest,
+  Context = unknown,
```

Comment:
I am thinking we might support `context` in `EventHandlerRequest` to avoid introducing new generic and increase portability of types

Reply:
I considered that initially. The problem is that this means that the _incoming_ request is typed as requiring or having a context, and will also make typing `$fetch` request considerably more complex.

For example, an `EventHandlerRequest<{ body: { id: string } }>` would not be assignable to `EventHandlerRequest<{ body: { id: string }, context: { other: true } }>` even though you can't _control_ the context from the thing initiating the _request_.

Reply:
I see (this also when locally trying to refactor now). I think once we have a chance before 1.8 release to introduce typed event, we might merge `body` and `query` types to the context so that `EventHandlerRequest<T>` the `T` generic is describing a validated event context (instead of two places)

Reply:
What do you mean by:
> we might merge `body` and `query` types to the context so that `EventHandlerRequest<T>` the `T` generic is describing a validated event context (instead of two places)

If it means what I think, then I would suggest we do not do this. I think it's important to separate out the incoming from outgoing. So the flow is:

* request comes in (requiring certain inputs)
* new 'validate' function runs, and _also_ returns values to context
* event is now typed with body/query and also context

As far as the request is concerned, the context typing is irrelevant, and it would make inference harder in Nitro and any other framework using EventHandler types if the context typing is merged with body/query.

Reply:
(context, is also part of incoming) i will explain better in out call :)